redaxo/bin/console

```diff
@@ -8,4 +8,28 @@ $REX['BACKEND_FOLDER'] = 'redaxo';
 
 chdir(dirname(__DIR__));
 
+// Check if posix is available
+if (!function_exists('posix_getuid')) {
+        echo "REDAXO console". PHP_EOL;
+        echo "-----------------------------------------------" . PHP_EOL;
+		echo "The posix extensions are required - see http://php.net/manual/en/book.posix.php" . PHP_EOL;
+		exit(1);
+	}
+// Check current user
+	$currentuser = posix_getpwuid(posix_getuid());
+	$webuser = posix_getpwuid(fileowner($REX['HTDOCS_PATH'].$REX['BACKEND_FOLDER']));
```

Comment:
Bei uns kommt das öfters vor, dass der SSH-Benutzer ein anderer ist als der Web-Benutzer.
Und der Webbenutzer bekommt nur Schreibrechte auf bestimmte Ordner, per `setfacl`. Der Owner bleibt also bei dem SSH-Benutzer.

Eventuell würde es dann zufällig wieder gehen, da der fileowner des Backend-Folders eben bei dem SSH-Benutzer bleibt, und somit dein Skript nicht meckern würde (obwohl es nicht der Webuser ist).
Insgesamt glaube ich aber, dass das Konstrukt so nicht allgemeingültig sinnvoll ist. Es kommt drauf an, wie man den Server konfiguriert hat, wie man deployed, und wie man die Schreibrechte dem Webuser gibt, etc.

Reply:
Ich denke, für solche Fälle sollte es einen Parameter geben um euer Szenario zu erzwingen. 
Ich gehe hier von einem "normalen" Hosting-Szenario aus. Mir ist es schon passiert, dass ich vergessen habe PHP mit dem korrekten User auszuführen und so die Dateirechte anschließend neu setzen musste.

Reply:
Alternativ sollte zumindest eine auffällige Warnung erscheinen.

Reply:
verstehe beide fälle. Das exit() finde ich auch etwas hart. Zudem gibt es nicht in jeder console `sudo`.

Ich bin eh am überlegen, ob es net sinn macht, die Logik innerhalb der Symfony console zu lösen, damit man u.a. das Styling hat, das Option handling und auch confirm() hat.

Reply:
Gerne. Ich wusste nicht wo es hin soll. Farbig wäre schick.